include/sys/util.h

```diff
@@ -119,6 +120,56 @@ static inline s64_t arithmetic_shift_right(s64_t value, u8_t shift)
 	return (value >> shift) | (sign_ext << (64 - shift));
 }
 
+/**
+ * @brief      Convert a single character into a hexadecimal nibble.
+ *
+ * @param[in]  c     The character to convert
+ * @param      x     The address of storage for the converted number.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int char2hex(char c, u8_t *x);
+
+/**
+ * @brief      Convert a single hexadecimal nibble into a character.
+ *
+ * @param[in]  c     The number to convert
+ * @param      x     The address of storage for the converted character.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int hex2char(u8_t x, char *c);
+
+/**
+ * @brief      Convert a binary array into string representation.
+ *
+ * @param      hex     Address of where to store the string representation.
+ * @param[in]  hexlen  Size of the storage area for string representation.
+ * @param[in]  buf     The binary array to convert
+ * @param[in]  buflen  The length of the binary array to convert
+ *
+ * @return     The length of the converted string, or 0 if an error occurred.
+ */
+size_t bin2hex(char *hex, size_t hexlen, const u8_t *buf, size_t buflen);
```

Comment:
IMHO for me it would be more intuitive if the buf param is the first and the hex param is the last because the function is called bin2hex(). Same for hex2bin(), the params should be swapped

Reply:
Good point, I also see that the versions that I removed in mesh/shell.c had it like that, and I didn't update it correctly.